integration-test/1491-generic-network-fixes.py

```diff
@@ -0,0 +1,4231 @@
+                'name': u'Mahendra Rajmarg',
+                'name:en': u'Mahendra Highway',
+                'name:ne': u'महेन्द्र राज्मार्ग्',
+                'name:pl': u'Mahendra',
+                'network': u'np:national',
+                'ref': u'H01',
+                'route': u'road',
+                'source': u'openstreetmap.org',
+                'type': u'route',
+                'wikidata': u'Q3595565',
+                'wikipedia': u'en:Mahendra Highway',
+            }, ways=[27030698]),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 27030698,
+                'network': u'NP:national',
+            })
+
+    def test_npregional(self):
+        import dsl
+
+        z, x, y = (16, 48796, 27744)
+
+        self.generate_fixtures(
+            dsl.is_in('NP', z, x, y),
+            # https://www.openstreetmap.org/way/352615958
+            dsl.way(352615958, dsl.tile_diagonal(z, x, y), {
+                'highway': u'primary',
+                'name': u'Birtamod — Chandragadhi',
+                'ref': u'F1',
+                'source': u'openstreetmap.org',
+                'surface': u'paved',
+            }),
+            dsl.relation(1, {
+                'distance': u'12.53 km',
+                'name': u'Birtamod — Chandragadhi',
+                'network': u'np:regional',
+                'ref': u'F1',
+                'route': u'road',
+                'source': u'openstreetmap.org',
+                'type': u'route',
+            }, ways=[352615958]),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 352615958,
+                'network': u'NP:regional',
+            })
+
+    def test_phnational(self):
+        import dsl
+
+        z, x, y = (16, 55524, 30719)
+
+        self.generate_fixtures(
+            dsl.is_in('PH', z, x, y),
+            # https://www.openstreetmap.org/way/460910438
+            dsl.way(460910438, dsl.tile_diagonal(z, x, y), {
+                'highway': u'trunk',
+                'int_ref': u'AH26',
+                'lanes': u'2',
+                'name': u'Maharlika Highway',
+                'nat_name': u'Daang Maharlika (LT)',
+                'official_name': u'Daang Maharlika Highway',
+                'ref': u'1;AH26',
+                'source': u'openstreetmap.org',
+                'surface': u'concrete',
+            }),
+            dsl.relation(1, {
+                'name': u'N1',
+                'network': u'ph:nhn',
+                'nhn:class': u'primary',
+                'ref': u'1',
+                'route': u'road',
+                'source': u'openstreetmap.org',
+                'type': u'route',
+            }, ways=[460910438]),
+            dsl.relation(2, {
+                'int_ref': u'AH26',
+                'int_ref:colour': u'blue',
+                'name': u'National Highway',
+                'name:en': u'National Highway',
+       
```

Comment:
Please add shield text test – we generally only have 1, 2, 3, 4, and 5-char artwork so this would not render unless the space were taken out of the ref, and possibly drop the `EN`? Needs research.

Reply:
Looks like that doesn't get signposted as `EN 230`, but as `N230`. I added that to the tests in 1603dc5 and did a deeper dive into the Portuguese road network in 46c3a02.

Basically, there are [several different network/shield types](https://en.wikipedia.org/wiki/Roads_in_Portugal#Current_classification_according_PRN_2000):

* `A` autoestradas, similar to motorways, so I've synthesized the network `PT:motorway`,
* `IP` itinerários principais, similar to primary routes, so I've synthesized the network `PT:primary`,
* `IC` itinerários complementares, similar to secondary routes, so I've synthesized the network `PT:secondary`,
* `EN` Estradas nacionais, national roads, for which I've reused the network `PT:national` (note that it's signed with just an `N`),
* `ER` Estradas regionais, regional roads, for which I've reused the network `PT:regional` (note that it's signed with just an `R`),
* `EM` Estradas municipais, municipal roads, so I've synthesized the network `PT:municipal` (note that these don't appear to be signed at all - I could not find an example),
* `VR` Via Rapida, seems to be a motorway class road only on Madeira, so I've synthesized the network `PT:rapid`,
* `VE` Via Expresso, seems to be a primary class road only on Madeira, so I've synthesized the network `PT:express`.